modules/internal/ChapelArray.chpl

```diff
@@ -5267,4 +5267,22 @@ module ChapelArray {
       return A;
     }
   }
+
+  // used for passing arrays to extern procs, e.g.
+  //   extern proc foo(X: []);
+  //   var A: [1..3] real;
+  //   foo(A);
+  // 'castToVoidStar' says whether we should cast the result to c_void_ptr
+  pragma "no doc"
+  proc chpl_arrayToPtr(arr: [], param castToVoidStar: bool = false) {
+    if (!chpl__isDROrDRView(arr)) then
+      compilerError("Only local rectangular arrays can be passed to an external routine argument with array type", 2);
```

Comment:
I don't think we are checking whether the array is local?

I'd also pass `2` as a named argument. `errorDepth=2`

Reply:
isDROrDRView should be synonymous with "local rectangular array" shouldn't it?  (in the future, we may have other local rectangular array domain maps, but I don't think we do today, nor a standard way to check "definitely local" vs. "possibly distributed".

Reply:
Yeah but what if it is local to another locale?

Reply:
Ah, now I see what you mean.  I was reading "local" as "local to a single locale".  Thanks for clarifying.

Reply:
https://github.com/chapel-lang/chapel/pull/16452/commits/38a2952693cfafa7992c2c42b0ae17a701c45efe should address both of these comments.